client/client.go

```diff
@@ -0,0 +1,421 @@
+package client
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/notary/trustmanager"
+	"github.com/endophage/gotuf"
+	"github.com/endophage/gotuf/data"
+	"github.com/endophage/gotuf/keys"
+	"github.com/endophage/gotuf/signed"
+	"github.com/endophage/gotuf/store"
+	"github.com/spf13/viper"
+)
+
+// Default paths should end with a '/' so directory creation works correctly
+const trustDir string = "/trusted_certificates/"
+const privDir string = "/private/"
+const tufDir string = "/tuf/"
+const rootKeysDir string = "/root_keys/"
+
+// ErrRepositoryNotExist gets returned when trying to make an action over a repository
+/// that doesn't exist
+var ErrRepositoryNotExist = errors.New("repository does not exist")
+
+// Client is the interface that defines the Notary Client type
+type Client interface {
+	ListPrivateKeys() []data.PrivateKey
+	GenRootKey(passphrase string) (*data.PublicKey, error)
+	GetRepository(gun string, baseURL string, transport http.RoundTripper) (Repository, error)
```

Comment:
Why use a RoundTripper? Does it afford some benefit to us over just doing an `http.Get`?